src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -137,22 +138,40 @@ private async Task HandleClusterTransaction(DocumentsOperationContext context, M
                 throw new ConcurrencyException($"Failed to execute cluster transaction due to the following issues: {string.Join(Environment.NewLine, errors)}");
             }
 
+            // wait for the command to be applied on this node
+            await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, result.Index);
+
             var array = new DynamicJsonArray();
             if (clusterTransactionCommand.DatabaseCommandsCount > 0)
             {
+#if DEBUG
```

Comment:
This needs to be removed, even in debug only.
It's fine if you have an error / assert, but console write line are not a good idea in our code except while trouble shooting.